src/channel_state.ts

```diff
@@ -603,6 +614,153 @@ export class ChannelState<
 
     return { removed: result.length < msgArray.length, result };
   };
+
+  /**
+   * @private
+   *
+   * Updates the `message.user` property on messages from given user,
+   * and returns the updates messages array
+   *
+   * @param {Array<ReturnType<ChannelState<AttachmentType, ChannelType, CommandType, EventType, MessageType, ReactionType, UserType>['formatMessage']>>} messages
+   * @param {UserResponse<UserType>} user
+   */
+  _updateUserMessages = (
+    messages: Array<
+      ReturnType<
+        ChannelState<
+          AttachmentType,
+          ChannelType,
+          CommandType,
+          EventType,
+          MessageType,
+          ReactionType,
+          UserType
+        >['formatMessage']
+      >
+    >,
+    user: UserResponse<UserType>,
+  ) =>
+    messages.map((m) => {
+      if (m.user?.id !== user.id) {
+        return m;
+      }
+
+      return {
+        ...m,
+        user,
+      };
+    });
+
+  /**
+   * Updates the message.user property with updated user object, for messages.
+   *
+   * @param {UserResponse<UserType>} user
+   */
+  updateUserMessages = (user: UserResponse<UserType>) => {
+    this.messages = this._updateUserMessages(this.messages, user);
+
+    for (const parentId in this.threads) {
+      this.threads[parentId] = this._updateUserMessages(this.threads[parentId], user);
+    }
+
+    this.pinnedMessages = this._updateUserMessages(this.pinnedMessages, user);
+  };
+
+  /**
+   * @private
+   *
+   * Marks messages from given user as deleted, and returns updated message list
+   *
+   * @param {Array<ReturnType<ChannelState<AttachmentType, ChannelType, CommandType, EventType, MessageType, ReactionType, UserType>['formatMessage']>>} messages
+   * @param {UserResponse<UserType>} user
+   * @param {boolean} hardDelete
+   */
+  _deleteUserMessages = (
+    messages: Array<
+      ReturnType<
+        ChannelState<
+          AttachmentType,
+          ChannelType,
+          CommandType,
+          EventType,
+          MessageType,
+          ReactionType,
+          UserType
+        >['formatMessage']
+      >
+    >,
+    user: UserResponse<UserType>,
+    hardDelete = false,
+  ) =>
+    messages.map((m) => {
+      if (m.user?.id !== user.id) {
+        return m;
+      }
+
+      if (hardDelete) {
+        /**
+         * In case of hard delete, we need to strip down all text, html,
+         * attachments and all the custom properties on message
+         */
+        return ({
+          cid: m.cid,
+          created_at: m.created_at,
+          deleted_at: user.deleted_at,
+          id: m.id,
+          latest_reactions: [],
+          mentioned_users: [],
+          own_reactions: [],
+          parent_id: m.parent_id,
+          reply_count: m.reply_count,
+          status: m.status,
+          thread_participants: m.thread_participants,
+          type: 'deleted',
+          updated_at: m.updated_at,
+          user: m.user,
```

Comment:
does it make sense to extract this into a default empty msg or sth?

Reply:
I did think about it as well, but it's not exactly empty message. There are certain properties that we retain and some we remove. So didn't make sense to me to extract it out.